sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlobClientBase.java

```diff
@@ -302,45 +306,61 @@ public BlobInputStream openInputStream(BlobInputStreamOptions options) {
-            ? new BlobRequestConditions() : options.getRequestConditions();
-        BlobAsyncClientBase client = this.client;
-
-        switch (consistentReadControl) {
-            case NONE:
-                break;
-            case ETAG:
-                // Target the user specified eTag by default. If not provided, target the latest eTag.
-                if (requestConditions.getIfMatch() == null) {
-                    requestConditions.setIfMatch(eTag);
-                }
-                break;
-            case VERSION_ID:
-                if (versionId == null) {
-                    throw logger.logExceptionAsError(
-                        new UnsupportedOperationException("Versioning is not supported on this account."));
-                } else {
-                    // Target the user specified version by default. If not provided, target the latest version.
-                    if (this.client.getVersionId() == null) {
-                        client = this.client.getVersionClient(versionId);
-                    }
+        com.azure.storage.common.ParallelTransferOptions parallelTransferOptions =
+            new com.azure.storage.common.ParallelTransferOptions().setBlockSizeLong((long) chunkSize);
+        BiFunction<BlobRange, BlobRequestConditions, Mono<BlobDownloadAsyncResponse>> downloadFunc =
+            (chunkRange, conditions) -> client.downloadWithResponse(chunkRange, null, conditions, false);
+        return ChunkedDownloadUtils.downloadFirstChunk(range, parallelTransferOptions, requestConditions, downloadFunc, true)
+            .flatMap(tuple3 -> {
+                BlobDownloadAsyncResponse downloadResponse = tuple3.getT3();
+                return FluxUtil.collectBytesInByteBufferStream(downloadResponse.getValue())
+                    .map(ByteBuffer::wrap)
+                    .zipWith(Mono.just(downloadResponse));
+            })
+            .map(tuple2 -> {
+                ByteBuffer initialBuffer = tuple2.getT1();
+                BlobDownloadAsyncResponse downloadResponse = tuple2.getT2();
+
+                BlobProperties properties = ModelHelper.buildBlobPropertiesResponse(downloadResponse).getValue();
+
+                String eTag = properties.getETag();
+                String versionId = properties.getVersionId();
+                BlobAsyncClientBase client = this.client;
+
+                switch (consistentReadControl) {
+                    case NONE:
+                        break;
+                    case ETAG:
+                        // Target the user specified eTag by default. If not provided, target the latest eTag.
+                        if (requestConditions.getIfMatch() == null) {
+                            requestConditions.setIfMatch(eTag);
+                        }
+                        break;
+                    case VERSION_ID:
+                        if (versionId == null) {
+                            throw logger.logExceptionAsError(
```

Comment:
nit: Change throws in the reactive stream to `Mono.error` returns. `FluxUtil.monoError` will be useful here